src/libs/Navigation/getTopmostCentralPaneName.ts

```diff
@@ -0,0 +1,19 @@
+import type {NavigationState, PartialState} from '@react-navigation/native';
+import SCREENS from '@src/SCREENS';
+
+// Get the name of topmost report in the navigation stack.
+function getTopmostCentralPaneName(state: NavigationState | PartialState<NavigationState>): string | undefined {
+    if (!state) {
+        return;
+    }
+
+    const topmostCentralPane = state.routes.filter((route) => typeof route !== 'number' && 'name' in route && route.name === SCREENS.SETTINGS_CENTRAL_PANE).at(-1);
+
+    if (!topmostCentralPane || typeof topmostCentralPane === 'number' || !('state' in topmostCentralPane)) {
+        return;
+    }
+
+    return topmostCentralPane.state?.routes.at(-1)?.name;
```

Comment:
```suggestion
    const topmostCentralPane = state.routes.filter((route) => route.name === SCREENS.SETTINGS_CENTRAL_PANE).at(-1);

    return topmostCentralPane?.state?.routes.at(-1)?.name;
```

What's with the typeof number checks? Can we just omit them?

Reply:
I've consulted with @blazejkustra, who migrated the navigation to TS, and those checks were added to get rid of some TypeScript warnings. Currently, we don't have a TS warning (at least here). So I think we can omit them.